stdlib/http/src/main/java/org/ballerinalang/net/http/FailoverClientHandshakeListener.java

```diff
@@ -0,0 +1,121 @@
+ */
+public class FailoverClientHandshakeListener extends WebSocketClientHandshakeListener {
+    private final WebSocketService wsService;
+    private final WebSocketClientConnectorListener clientConnectorListener;
+    private final boolean readyOnConnect;
+    private final ObjectValue webSocketClient;
+    private CountDownLatch countDownLatch;
+    private static final Logger logger = LoggerFactory.getLogger(FailoverClientHandshakeListener.class);
+
+    FailoverClientHandshakeListener(ObjectValue webSocketClient, WebSocketService wsService,
+                                    WebSocketClientConnectorListener clientConnectorListener, boolean readyOnConnect,
+                                    CountDownLatch countDownLatch) {
+        super(webSocketClient, wsService, clientConnectorListener, readyOnConnect, countDownLatch);
+        this.webSocketClient = webSocketClient;
+        this.wsService = wsService;
+        this.clientConnectorListener = clientConnectorListener;
+        this.readyOnConnect = readyOnConnect;
+        this.countDownLatch = countDownLatch;
+    }
+
+    @Override
+    public void onSuccess(WebSocketConnection webSocketConnection, HttpCarbonResponse carbonResponse) {
+        //using only one service endpoint in the client as there can be only one connection.
+        webSocketClient.set(WebSocketConstants.CLIENT_RESPONSE_FIELD,
+                HttpUtil.createResponseStruct(carbonResponse));
+        ObjectValue webSocketConnector = BallerinaValues.createObjectValue(PROTOCOL_PACKAGE_HTTP,
+                WebSocketConstants.WEBSOCKET_CONNECTOR);
+        WebSocketOpenConnectionInfo connectionInfo = new WebSocketOpenConnectionInfo(
+                wsService, webSocketConnection, webSocketClient);
+        webSocketConnector.addNativeData(WebSocketConstants.NATIVE_DATA_WEBSOCKET_CONNECTION_INFO, connectionInfo);
+        clientConnectorListener.setConnectionInfo(connectionInfo);
+        webSocketClient.set(WebSocketConstants.CLIENT_CONNECTOR_FIELD, webSocketConnector);
+        FailoverContext failoverConfig = (FailoverContext) webSocketClient.getNativeData(FAILOVER_CONFIG);
+        if (failoverConfig.isConnectionMade()) {
+            webSocketClient.set(WebSocketConstants.LISTENER_ID_FIELD, webSocketConnection.getChannelId());
+        } else {
+            WebSocketUtil.populateEndpoint(webSocketConnection, webSocketClient);
+        }
+        // Read the next frame when readyOnConnect is true in first connection or after the first connection
+        if (failoverConfig.isConnectionMade() || readyOnConnect) {
+            webSocketConnection.readNextFrame();
+        }
+        // Following these are created for future connection
+        // Check whether the config has retry config or not
+        // It has retry config, set these variable to default variable
+        FailoverContext failoverContext = ((FailoverContext) webSocketClient.getNativeData(FAILOVER_CONFIG));
```

Comment:
You have created the same object on line `76`
`FailoverContext failoverConfig = (FailoverContext) webSocketClient.getNativeData(FAILOVER_CONFIG);`
and i think it is wrong to wrong and misleading to name this `failoverConfig`